client-rpc/src/wallet_rpc.rs

```diff
@@ -237,15 +237,15 @@ mod tests {
             Ok(Vec::new())
         }
 
-        fn transaction(&self, _: &TxId) -> Result<Option<Tx>> {
-            Ok(Some(Tx {
+        fn transaction(&self, _: &TxId) -> Result<Option<Transaction>> {
+            Ok(Some(Transaction::TransferTransaction(Tx {
```

Comment:
how about just using Result<Tx>

Reply:
It's already using `Result<Option<...>>`?

Reply:
Result with Option seems redundant

Reply:
This function is mainly used by UI. @calvinaco any comments?

Reply:
@leejw51crypto it depends on the semantics -- it could be that `Ok(None)` indicates there wasn't any transaction found, but everything else was OK (connection was OK, TXID was a correct length, I/O was OK...)

Reply:
But it could be that one of the errors could be "TransactionNotFound", and the whole return type could be `Result<Transaction>` instead... probably simpler -- perhaps in a different PR?